### PR TITLE
Meal tests

### PR DESCRIPTION
**Pull request by Benjaminpjacobs**
This PR adds tests for the Meal object as well as GrandTotalNodes and TableNodes objects. It implements Sinon for mocking particular callbacks and has separate stub files for the individual tests.

### CODE REVIEW

File: lib/grand_total_nodes.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class GrandTotalNodes {
+    constructor(tables) {
+        this.goal = tables.find('#goal-calories')
+        this.total = tables.find('#total-calories')
+        this.remaining = tables.find('#remaining-calories')
+        this.tableTotals = tables.find('.total-calories')
+    }
+
```
Comment (podoglyph) [inline lib/grand_total_nodes.js:8]: Nice. This seems like a really good way to break out more functionality into separate pieces, as well as add some efficiency and readability.

**Review by podoglyph — APPROVED** [review]

**Comment by podoglyph** [comment]
Good test coverage for the new node objects you've created. Nice work.

File: test/meal_test.js [hunk 2]
```diff
@@ -1,25 +1,78 @@
...
+        assert.equal(table.find('.remaining-calories').text(), 0)
+
+        assert(table.find('.negative').length === 0)
+        assert(table.find('.negative'))
+
+        assert.equal(totalTable.find('#goal-calories').text(), 2000)
+        assert.equal(totalTable.find('#total-calories').text(), 400)
+        assert.equal(totalTable.find('#remaining-calories').text(), 1600)
+
+        sinon.assert.calledTwice(mealCalcTotalStub)
```
Comment (case-eee) [inline test/meal_test.js:65]: 👍

File: test/stubs/meal_stub.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const Meal = require("../../lib/meal.js")
+
+module.exports = new Meal({
```
Comment (case-eee) [inline test/stubs/meal_stub.js:3]: 👍

File: lib/grand_total_nodes.js (hunk 1 above)
Comment (case-eee) [reply lib/grand_total_nodes.js:8]: agree!

File: test/food_test.js [hunk 4]
```diff
@@ -78,64 +78,64 @@ describe('test food object', function() {
...
-            .then(function(foods) {
-                let diff = originalFoodList - foods.length
-                assert.equal(diff, 1)
-            });
-    });
-
-
-});
\ No newline at end of file
+// test.describe('testing my foods', function() {
```
Comment (case-eee) [inline test/food_test.js:81]: are y'all working on uncommenting these to get these working again?

**Review by case-eee — COMMENTED** [review]
@Benjaminpjacobs @podoglyph tests are looking solid!

**Pull request merged**